app/controllers/documentText.ts

```diff
@@ -21,6 +21,27 @@ export default function(app) {
 
       const revisionId = $routeSegment.$routeParams.revisionId;
 
+      function getAccessiblePdfUrl(documentRevision) {
+        const userHasAccess = documentRevision.isOpenAccess;
+        if (!userHasAccess) {
+          throw new Error('You currently have no access to the PDF.');
+        }
```

Comment:
Is throwing appropriate here? The user won't notice this. Below we use a notification for error handling...